### PR TITLE
ROC-3200 Calculate interest using issuedOn date

### PR DESCRIPTION
**Pull request by robertparkinson**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-3200

### Change description ###
Interest calculation was incorrectly using createdAt date rather than issuedOn date. This PR is to update to use the correct date

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/InterestProviderTest.java [hunk 1]
```diff
@@ -25,7 +24,7 @@
     private Interest interest;
     private InterestDate interestDate;
     private BigDecimal claimAmount;
-    private LocalDateTime submittedOn;
+    private LocalDate submittedOn;
```
Comment (geek-soft) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/InterestProviderTest.java]: Please rename it to `issuedOn`, submitted on means slightly different thing IMO.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/InterestContentProvider.java [hunk 2]
```diff
@@ -29,7 +29,7 @@ public InterestContent createContent(
         Interest interest,
         InterestDate interestDate,
         BigDecimal claimAmount,
-        LocalDateTime submittedOn) {
+        LocalDate submittedOn) {
```
Comment (geek-soft) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/content/InterestContentProvider.java]: Please rename it to issuedOn, submitted on means slightly different thing IMO.

**Comment by geek-soft** [comment]
Also look at `uk/gov/hmcts/cmc/claimstore/services/staff/content/countycourtjudgment/AmountContentProvider.java:43` (something I highlighted yesterday) - the way dates are calculated for display in CCJ copy differs from the way dates are calculated for interest calculation in the same PDF

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculator.java [hunk 3]
```diff
@@ -59,18 +59,23 @@ public static BigDecimal asFraction(BigDecimal interestRate) {
 
     private static BigDecimal calculateTotalAmount(Claim claim, LocalDate toDate) {
         ClaimData data = claim.getClaimData();
+        LocalDate afterIssuedOnDate = toDate;
+
+        if (toDate.isBefore(claim.getIssuedOn())) {
+            afterIssuedOnDate = claim.getIssuedOn();
```
[the remark this answers is not in the document]
Comment (geek-soft) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculator.java]: The comment is still pending.

File: domain-model/src/test/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculatorTest.java [hunk 4]
```diff
@@ -99,7 +99,7 @@ public void totalTillDateOfIssueShouldReturnClaimAmountPlusFeeWhenNoInterest() {
     @Test
     public void totalTillTodayShouldReturnCalculatedTotalAmountPlusInterestWhenCalculateFromSubmission() {
```
Comment (geek-soft) [inline domain-model/src/test/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculatorTest.java]: Please update description to highlight that interest is calculated from issue date, not submission date.

File: domain-model/src/test/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculatorTest.java [hunk 5]
```diff
@@ -109,7 +109,61 @@ public void totalTillTodayShouldReturnCalculatedTotalAmountPlusInterestWhenCalcu
...
+            )
+            .withIssuedOn(LocalDate.now().plusDays(1))
+            .build();
+
+        assertThat(TotalAmountCalculator.totalTillToday(claimStandardInterest))
+            .isEqualTo(Optional.of(format(new BigDecimal("60"))));
+    }
+
+    @Test
+    public void totalTillTodayWhenClaimIssuedOnIsDayAfterTomorrow() {
```
Comment (geek-soft) [inline domain-model/src/test/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculatorTest.java]: I think we should replace this with the scenario where case was issued yesterday. In that case I would expect total amount to interest for one day.

File: domain-model/src/test/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculatorTest.java (hunk 4 above)
Comment (sabahirfan) [reply domain-model/src/test/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculatorTest.java]: done

File: domain-model/src/test/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculatorTest.java (hunk 5 above)
Comment (sabahirfan) [reply domain-model/src/test/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculatorTest.java]: added yesterday's scenario as well.

File: domain-model/src/test/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculatorTest.java [hunk 6]
```diff
@@ -109,10 +109,82 @@ public void totalTillTodayShouldReturnCalculatedTotalAmountPlusInterestWhenCalcu
                     .build()
             ).build();
 
-        assertThat(TotalAmountCalculator.totalTillDateOfIssue(claimNoInterest))
+        assertThat(TotalAmountCalculator.totalTillDateOfIssue(claimStandardInterest))
             .isEqualTo(Optional.of(format(new BigDecimal("60"))));
     }
 
+    @Test
+    public void totalTillTodayWhenClaimIssuedOnIsToday() {
```
Comment (geek-soft) [inline domain-model/src/test/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculatorTest.java]: I would love rename the test case to say what is expected e.g. `totalTillTodayShouldNotIncludeInterestWhenClaimIssuedOnIsToday`.

File: domain-model/src/test/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculatorTest.java [hunk 7]
```diff
@@ -109,10 +109,82 @@ public void totalTillTodayShouldReturnCalculatedTotalAmountPlusInterestWhenCalcu
...
+            )
+            .withIssuedOn(LocalDate.now())
+            .build();
+
+        assertThat(TotalAmountCalculator.totalTillToday(claimStandardInterest))
+            .isEqualTo(Optional.of(format(new BigDecimal("60"))));
+    }
+
+    @Test
+    public void totalTillTodayWhenClaimIssuedOnIsTomorrow() {
```
Comment (geek-soft) [inline domain-model/src/test/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculatorTest.java]: The same way that method should say `totalTillTodayShouldNotIncludeInterestWhenClaimIssuedOnIsTomorrow`.

File: domain-model/src/test/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculatorTest.java (hunk 5 above)
Comment (geek-soft) [reply domain-model/src/test/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculatorTest.java]: I would be happy to remove that scenario as this tests the same execution path as previous test.

File: domain-model/src/test/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculatorTest.java [hunk 8]
```diff
@@ -109,10 +109,82 @@ public void totalTillTodayShouldReturnCalculatedTotalAmountPlusInterestWhenCalcu
...
+            )
+            .withIssuedOn(LocalDate.now().plusDays(2))
+            .build();
+
+        assertThat(TotalAmountCalculator.totalTillToday(claimStandardInterest))
+            .isEqualTo(Optional.of(format(new BigDecimal("60"))));
+    }
+
+    @Test
+    public void totalTillTodayWhenClaimIssuedOnIsYesterday() {
```
Comment (geek-soft) [inline domain-model/src/test/java/uk/gov/hmcts/cmc/domain/amount/TotalAmountCalculatorTest.java]: The same way that should say `totalTillTodayShouldIncludeOneDayInterestWhenClaimIssuedOnIsYesterday`

**Pull request merged**